src/main/groovy/com/netflix/gradle/plugins/application/OspackageApplicationSpringBootPlugin.groovy

```diff
@@ -54,85 +53,95 @@ class OspackageApplicationSpringBootPlugin implements Plugin<Project> {
-                    if (project.application.mainClassName == null) {
-                        project.application.mainClassName = mainClass.getOrNull() // Fail only when startScripts runs
+                    if (!mainClass.isPresent()) {
+                        try {
+                            mainClass.set(project.application.mainClass.isPresent() ? project.application.mainClass.get() : project.application.mainClassName)
+                        } catch (Exception ignore) {
+                        }
+                    }
+                    if (GradleVersion.current().baseVersion < GradleVersion.version('6.4').baseVersion) {
+                        if (project.application.mainClassName == null) {
+                            project.application.mainClassName = mainClass.getOrNull()
+                            // Fail only when startScripts runs
+                        }
+                    } else {
+                        project.application.mainClass.convention(mainClass)
                     }
-                } else {
-                    project.application.mainClass.convention(mainClass)
                 }
-            }
 
-            // Workaround for https://github.com/gradle/gradle/issues/16371
-            if (GradleVersion.current().baseVersion >= GradleVersion.version('6.4').baseVersion) {
-                project.tasks.named(ApplicationPlugin.TASK_START_SCRIPTS_NAME).configure {
-                    doFirst {
-                        if (!project.application.mainClass.isPresent()) {
-                            throw new GradleException("mainClass should be configured in order to generate a valid start script. i.e. mainClass = 'com.netflix.app.MyApp'")
+                // Workaround for https://github.com/gradle/gradle/issues/16371
+                if (GradleVersion.current().baseVersion >= GradleVersion.version('6.4').baseVersion) {
+                    project.tasks.named(ApplicationPlugin.TASK_START_SCRIPTS_NAME).configure {
+                        doFirst {
+                            if (!project.application.mainClass.isPresent()) {
+                                throw new GradleException("mainClass should be configured in order to generate a valid start script. i.e. mainClass = 'com.netflix.app.MyApp'")
+                            }
                         }
                     }
                 }
-            }
-        } else {
-            project.afterEvaluate {
-                project.tasks.named(DistributionPlugin.TASK_INSTALL_NAME).configure {
-                    it.dependsOn('bootRepackage')
-                }
-                project.tasks.named(ApplicationPlugin.TASK_START_SCRIPTS_NAME).configure { CreateStartScripts createStartScripts ->
-                    createStartScripts.mainClassName = 'org.springframework.boot.loader.JarLauncher'
-                }
+            } else {
+                project.logger.info("Spring Boot 1 detected")
```

Comment:
👍 helpful logging indeed, especially for this much older and thus unexpected version